### PR TITLE
Add migration types blog post

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/_posts/2020-12-22-migration-types.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Each of these approaches migrates user account data into FusionAuth from one or more legacy systems. Let's examine each approach in more detail to help you make the best decision for your use case.
+
+## The Big Bang Migration
+
+With a big bang migration, you are moving all your users at one time. How long this takes varies depending on the amount of data you are moving, but there is a single cutover. For small systems it may be minutes, for bigger systems it might last hours or even days. 
+
+The basic steps for this approach are:
+
+* Map user attributes from the legacy system to FusionAuth. As part of this, Decide which applications and tenants will need to be created in FusionAuth beforehand.
```
Comment (dpolstra) [inline site/_posts/2020-12-22-migration-types.md:33]: as part of this, ~D~decide

File: site/_posts/2020-12-22-migration-types.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* When this user logs in subsequently, FusionAuth responds; it is now the system of record, the user has been migrated, and the old system no longer has that particular user's information.
+
+This approach has a number of benefits. You are only migrating at the time a user authenticates. Therefore if there are issues, the blast radius is smaller and limited to the user authenticating. You have the user's password (they're providing it when they are authenticating), so you can upgrade their password hash to use a different algorithm transparently. This approach sidesteps any complexities around porting over bespoke hashing algorithms as well. 
+
+Application cutover is, all other things equal, simpler. You aren't moving any account data, which can often take some time. Instead you are only switching where users authenticate. In addition, you can scrub your user base of inactive users with little effort. Or, before the migration, you can contact them and encourage them to log in your applications, possibly regaining their interest.
```
Comment (dpolstra) [inline site/_posts/2020-12-22-migration-types.md:93]: and encourage them to log in **_to_** your application

File: site/_posts/2021-03-23-migration-types.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+## Slow Migration
+
+This approach is the logical endpoint of segment by segment migration. What if you made each chunk of accounts to be migrated as small as possible. Well, you end up with each segment being a single user. 
+
+With the slow migration approach, you need to: 
+
+* Map your user attributes from the old system to the new system. 
+* Set up a connection between the legacy system and FusionAuth. This could be an API call or some other way to pass authentication information from FusionAuth to the old system, and get back an authentication response.
+* Modify the application or applications to send users to FusionAuth for authentication. Test these changes before rolling them out, including that the connection created above works. If you previously used OIDC or SAML to connect to the legacy system, this effort may be minimal.
+* At this point, FusionAuth receives all auth requests. It delegates the initial authentication for each user to the legacy system.
```
Comment (jobannon) [inline site/_posts/2021-03-23-migration-types.md:87]: is shorthand "auth" ok here?

File: site/_posts/2021-03-23-migration-types.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Application cutover is, all other things equal, simpler. You aren't moving any account data, which can often take some time. Instead you are only switching where users authenticate. In addition, you can scrub your user base of inactive users with little effort. Or, before the migration, you can contact them and encourage them to log in to your applications, possibly regaining their interest.
+
+However, a slow migration isn't always the best solution. Some things to consider include:
+
+* You are passing a plaintext password or other credential from FusionAuth to the legacy system. Make sure you take care of this precious cargo using TLS at a minimum. If possible, keep it from travelling over the internet by placing FusionAuth and the legacy system on the same network.
```
Comment (jobannon) [inline site/_posts/2021-03-23-migration-types.md:97]: ...cargo using TLS (or some other packet layer security protocol) ....(might not be a thing) 😄

**Review by jobannon — COMMENTED** [review]
All stylistic/preference, really.  Great article!

**Pull request merged**